Resources/Storage/OccupancyPlugins/WLCGAccountingJson.py

```diff
@@ -0,0 +1,101 @@
+  """ .. class:: WLCGAccountingJson
+
+  Occupancy plugin to return the space information given by WLCG Accouting Json
+  """
+  def __init__(self, se):
+    self.se = se
+    self.log = se.log.getSubLogger('WLCGAccountingJson')
+    self.name = self.se.name
+
+  def getOccupancy(self, **kwargs):
+    """ Returns the space information given by WLCG Accouting Json
+
+        :returns: S_OK with dict (keys: SpaceReservation, Total, Free)
+    """
+    occupancyLFN = kwargs['occupancyLFN']
+
+    if not occupancyLFN:
+      return S_ERROR("Failed to get occupancyLFN")
+
+    tmpDirName = tempfile.mkdtemp()
+    filePath = os.path.join(tmpDirName, os.path.basename(occupancyLFN))
+
+    for storage in self.se.storages:
+      try:
+        ctx = gfal2.creat_context()
+        params = ctx.transfer_parameters()
+        params.overwrite = True
+        res = storage.updateURL(occupancyLFN)
+        if not res['OK']:
+          continue
+        occupancyURL = res['Value']
+        ctx.filecopy(params, occupancyURL, 'file://' + filePath)
+
+      except gfal2.GError as e:
+        detailMsg = "Failed to copy file %s to destination url %s: [%d] %s" % (
+            occupancyURL, filePath, e.code, e.message)
+        self.log.debug("Exception while copying", detailMsg)
+        continue
+
+      else:
+        break
+
+    if not os.path.isfile(filePath):
+      return S_ERROR('No WLCGAccountingJson file of %s is downloaded.' % (self.name))
+
+    with open(filePath, 'r') as path:
+      occupancyDict = json.load(path)
+
+    # delete temp dir
+    shutil.rmtree(tmpDirName)
+
+    try:
+      storageShares = occupancyDict['storageservice']['storageshares']
+    except KeyError as e:
+      return S_ERROR('Could not find %s key in %s at %s' % (str(e), occupancyLFN, self.name))
+
+    spaceReservation = self.se.options.get('SpaceReservation')
+
+    # get storageshares in WLCGAccountingJson file
+    storageSharesSR = None
+    if spaceReservation:
+      for storageshare in storageShares:
+        if storageshare.get('name') == spaceReservation:
+          storageSharesSR = storageshare
+          break
+    else:
+      self.log.debug(
+          'Could not find SpaceReservation in CS, and get storageShares and spaceReservation from WLCGAccoutingJson.')
+      shareLen = []
+      for storage in self.se.storages:
+        basePath = storage.getParameters()['Path']
+        for share in storageShares:
+          shareLen.append((share, len(os.path.commonprefix([share['path'][0], basePath]))))
+      storageSharesSR = max(shareLen, key=lambda x: x[1])[0]
+      spaceReservation = storageSharesSR.get('name')
+
+    sTokenDict = {}
+    sTokenDict['SpaceReservation'] = spaceReservation
+    try:
+      sTokenDict['Total'] = storageSharesSR['totalsize']
+      sTokenDict['Free'] = sTokenDict['Total'] - storageSharesSR['usedsize']
+    except KeyError as e:
+      return S_ERROR('Could not find %s key in %s storageshares' % (str(e), spaceReservation))
```

Comment:
And here I would do:
```suggestion
      self.log.error("Issue finding Total or Free space left", 'Could not find %s key in %s storageshares' % (repr(e), spaceReservation))
      return S_ERROR("Issue finding Total or Free space left")
```

Reply:
Not sure on that one. I would not for with  `error` because it would be up to higher level to report the error, maybe `debug` ?

Reply:
Good point. But then the information “could not find the key” would not be visible in usual cases if it’s debug...
Why not put the phrase in S_ERROR?

Reply:
OK with me to add it in the S_ERROR message. Maybe you could also add a meaningful error number, lke `ENOMSG` maybe ?